src/app/page.tsx

```diff
@@ -1,47 +1,55 @@
-import Image from "next/image";
+"use client";
+
+import { PairsList } from "./PairsList";
+import { createContext, useEffect, useState } from "react";
+import * as adex from "alphadex-sdk-js";
+import { PairInfo } from "./PairInfo";
+
+adex.init(); //Connect to alphadex websocket
+
+let initial_staticState = new adex.StaticState(adex.clientState.internalState);
+
+export const AdexStateContext = createContext(initial_staticState);
 
 // more components here: https://daisyui.com/components/
 
 export default function Home() {
+  const [adexState, setAdexState] = useState(initial_staticState);
+  const [hydrated, setHydrated] = useState(false);
```

Comment:
Hydration only makes sense when using SSR, in our app we should be able to go without it theoretically. cc @fliebenberg what do you think?

Reply:
Initially I left this out, but it caused hydration errors on refresh. 

Is SSR feasible given that the client will be selecting pairs, moving graphs around, creating orders etc?

Reply:
I mean SSR (and therefore hydration) only makes sense if we have a server, we won't. I'll try to find a solution with #4

Reply:
Although SSG (static site generation) is also a kind of prerendering... and maybe I'm wrong here about hydration 🤔

Reply:
Ok, I consulted an independent React developer (pro level): this setup should be ok, if `initial_staticState` will always be the same (otherwise initial state should be `null` and all initialization should be done inside `useEffect` (so that it would always execute client-side (i.e. not during SSG). If not it will give us some errors in the future when it changes. Given that this state contains tokens and pairs, I bet it will change when more tokens appear. cc @fliebenberg

Reply:
Not even errors - undefined behavior. 
The issue stems from how React works with hydration. During hydration, React needs to reconcile the server-rendered HTML with the client-side state (they need to be identical, if they are not identical in production it's undefined behavior, and the page may start to glitch). 
`useEffect` runs after the render is committed to the screen, meaning it runs after the initial render and after every update. But if you put to console `console.log("initial_staticState", initial_staticState);` just below the initialization (and before the rendering `Home()` function), you can see there are already `tokensMap` and `pairsMap`. I suspect they might change in the future (without changing the library version). But I don't know for sure.

Reply:
The initial static state should not change (it is specifically a static state that does not change if things change on AlphaDEX. The state is only updated when a new static state is assigned in the subscription that happens in the useEffect. So I think it should be fine.
Note that all this might change after I refactor the SDK, but we will deal with that at that time.